actors/evm/src/lib.rs

```diff
@@ -116,14 +124,15 @@ impl EvmContractActor {
 
     pub fn invoke_contract<BS, RT>(
         rt: &mut RT,
+        caller: &EthAddress,
```

Comment:
https://github.com/filecoin-project/FIPs/discussions/459#discussioncomment-3666005

Reply:
wait, so evwryrhing that interacts with an evm contract must have an eth address?

This looks totally wrong, are we ok with ID addrs encoded as eth addresses?

Reply:
> wait, so everything that interacts with an evm contract must have an eth address?

Yes, even if it's just an ID encoded as an eth address.

> This looks totally wrong, are we ok with ID addrs encoded as eth addresses?

Yes.

Reply:
We're now going to be pulling this out of the state-tree.